Importing Libraries


In [ ]:
import pandas as pd
import numpy as np
import seaborn as sns
import numpy as np

Reading data set from drive

In [ ]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [ ]:
data = pd.read_excel('/content/drive/MyDrive/DSA/ Activity Data sets/Week 6/Dataset_w6.xlsx')

Checking dataset

In [ ]:
data.head(15)

country  ... International trade: Imports (million US$)
0           Afghanistan  ...                                       3568
1               Albania  ...                                       4669
2               Algeria  ...                                      47091
3                Angola  ...                                       8790
4   Antigua and Barbuda  ...                                        491
5             Argentina  ...                                      55610
6               Armenia  ...                                       3230
7             Australia  ...                                     189406
8               Austria  ...                                     149299
9            Azerbaijan  ...                                       8532
10              Bahamas  ...                                       2904
11              Bahrain  ...                                      14749
12           Bangladesh  ...                                      52624
13             Barbados  ...                                       1621
14              Belarus  ...                                      27464

[15 rows x 7 columns]

In [ ]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 196 entries, 0 to 195
Data columns (total 7 columns):
 #   Column                                      Non-Null Count  Dtype  
---  ------                                      --------------  -----  
 0   country                                     196 non-null    object 
 1   Region                                      196 non-null    object 
 2   Surface area (km2)                          196 non-null    int64  
 3   Population in thousands (2017)              196 non-null    int64  
 4   GDP per capita (current US$)                196 non-null    float64
 5   Agricultural production index               196 non-null    int64  
 6   International trade: Imports (million US$)  196 non-null    int64  
dtypes: float64(1), int64(4), object(2)
memory usage: 10.8+ KB


In [ ]:
data.isna().sum()

country                                       0
Region                                        0
Surface area (km2)                            0
Population in thousands (2017)                0
GDP per capita (current US$)                  0
Agricultural production index                 0
International trade: Imports (million US$)    0
dtype: int64

In [ ]:
data.columns

Index(['country', 'Region', 'Surface area (km2)',
       'Population in thousands (2017)', 'GDP per capita (current US$)',
       'Agricultural production index ',
       'International trade: Imports (million US$)'],
      dtype='object')

Q1: Find out how many countries are there in the dataset.

In [ ]:
data['country'].count()

196

There are 196 countries in the dataset()

# Q2: Find out how many countries with respect to the region are available in the dataset

In [ ]:
data['Region'].value_counts()

WesternAsia          18
EasternAfrica        17
Caribbean            16
WesternAfrica        15
SouthAmerica         12
SouthernEurope       12
South-easternAsia    11
EasternEurope        10
NorthernEurope       10
MiddleAfrica          9
SouthernAsia          9
CentralAmerica        8
WesternEurope         7
EasternAsia           7
Polynesia             5
CentralAsia           5
NorthernAfrica        5
SouthernAfrica        5
Melanesia             5
NorthernAmerica       4
Micronesia            4
Oceania               2
Name: Region, dtype: int64

Western Asia region has more countries and Oceania has the least number of countries

Q3: On average which region has the highest GDP per capita (current US $)


In [ ]:
data.groupby('Region')['GDP per capita (current US$)'].mean().sort_values()

Region
WesternAfrica         1013.140000
EasternAfrica         2088.852941
SouthernAsia          2968.400000
MiddleAfrica          3928.833333
NorthernAfrica        3934.960000
SouthernAfrica        4191.900000
CentralAsia           4329.880000
Micronesia            5008.300000
CentralAmerica        6348.137500
SouthAmerica          8499.775000
Polynesia             8686.520000
Melanesia             9261.960000
EasternEurope         9292.490000
South-easternAsia    10299.027273
SouthernEurope       14462.741667
Caribbean            15135.562500
WesternAsia          17013.938889
EasternAsia          27967.614286
NorthernEurope       42098.220000
Oceania              44823.250000
WesternEurope        55387.128571
NorthernAmerica      57659.000000
Name: GDP per capita (current US$), dtype: float64

Northern America region has the highest GDP per Capita in $

Q3: On average which region has the highest International trade: Imports (million US$)


In [ ]:
data.groupby('Region')['International trade: Imports (million US$)'].mean().sort_values()

Region
Micronesia               99.000000
Polynesia               432.000000
Melanesia              2896.400000
Caribbean              3057.312500
MiddleAfrica           4381.222222
EasternAfrica          4726.176471
WesternAfrica          4898.333333
CentralAsia           12063.400000
SouthernAfrica        18005.600000
SouthAmerica          32885.666667
NorthernAfrica        35034.400000
WesternAsia           46025.555556
CentralAmerica        57031.500000
SouthernAsia          58199.333333
SouthernEurope        76666.666667
EasternEurope         85243.900000
South-easternAsia     98324.636364
Oceania              112914.500000
NorthernEurope       113407.200000
WesternEurope        407498.428571
EasternAsia          451995.000000
NorthernAmerica      663552.250000
Name: International trade: Imports (million US$), dtype: float64

Northern America region has the highest International trade : Imports in $

Q4: Find out which region has more consistent with respect to GDP per capita (current US$)


In [ ]:
data.groupby('Region')['GDP per capita (current US$)'].std().sort_values()

Region
WesternAfrica          815.095399
NorthernAfrica         974.706575
SouthernAfrica        2177.551120
SouthernAsia          2855.708179
EasternAfrica         3716.131770
CentralAsia           4152.215490
CentralAmerica        4180.440883
SouthAmerica          4273.707394
Micronesia            4820.669694
MiddleAfrica          5274.846281
EasternEurope         5415.488739
Polynesia             6956.250286
Oceania               9233.329638
SouthernEurope        9541.703216
Melanesia            13855.842686
Caribbean            14579.371326
South-easternAsia    16311.810656
WesternAsia          18812.151459
NorthernEurope       20671.030185
WesternEurope        24770.467728
NorthernAmerica      25749.619445
EasternAsia          27485.120923
Name: GDP per capita (current US$), dtype: float64

Since the standard deviation for Western Africa is (815.095399) small compared to others, it is more consistent.